numpy/core/tests/test_simd.py

```diff
@@ -0,0 +1,333 @@
+# FIXME: written as pytest tests.
+import unittest
+from numpy.core._simd import targets
+
+class _TestSIMD(object):
+    """
+    Please avoid the use of numpy.testing since NPYV intrinsics
+    may be involved in their functionality.
+    """
+    npyv = None
+    sfx  = None
+    def __init__(self, methodName="runTest"):
+        unittest.TestCase.__init__(self, methodName)
+        unittest.TestCase.maxDiff = 1024
+
+    def test_memory(self):
+        data = self._data()
+        simd_load = self.load(data)
+        self.assertEqual(simd_load, data)
```

Comment:
`AttributeError: assertEqual`. Why not just `assert simd_load == data`?

Reply:
because PyPy using tuple as the default type for sequence protocol rather than a list,
but later I fixed this issue from the root so I guess using assert will be fine.
the fix here:
https://github.com/numpy/numpy/blob/6801b58ef3e101ac3c06b7804d2315c9b03bdcd5/numpy/core/src/_simd/_simd_inc_vector.h#L79-L86
and
https://github.com/numpy/numpy/blob/6801b58ef3e101ac3c06b7804d2315c9b03bdcd5/numpy/core/src/_simd/_simd_inc_vector.h#L67-L72

Reply:
> assert keyword is optimized out

I don't think this is a valid concern, the pytest assertion rewriter presumably works just fine in optimized mode.

Reply:
it looks nicer now, thanks!